src/search_unused.rs

```diff
@@ -343,72 +343,72 @@ pub(crate) fn find_unused(
 
     // TODO extend to dev dependencies + build dependencies, and be smarter in the grouping of
     // searched paths
-    let dependencies_names: Vec<_> = if let Some(resolve) = analysis
+    // Maps dependency name (the name of the key in the Cargo.toml dependency
+    // table, can have dashes, not necessarily the name in the crate registry)
+    // to crate name (extern crate, snake case)
+    let dependencies: BTreeMap<String, String> = if let Some(metadata) = analysis
         .metadata
         .as_ref()
-        .and_then(|metadata| metadata.resolve.as_ref())
+        .filter(|metadata| metadata.resolve.is_some())
```

Comment:
To avoid the `unwrap()` on the next line, we can probably zip the metadata along the resolve:

```rust
if let Some((metadata, resolve)) = analysis
  .metadata
  .as_ref()
  .filter_map(|metadata| metadata.resolve.as_ref().and_then(|resolve| (metadata, resolve)) { 
```

or something like that?